### ISSUE TITLE
Request:Soft System Link

### ISSUE THREAD

**Issue by eregen** [ddvk/remarkable-hacks#487]
I have many pdfs of technical textbooks and research papers and I like to organize them by topic or keywords. So ideally I would like to structure them into files where all relevant texts can be stored. But this can apply to more than one folder. Trouble is they take up a lot of storage space, so holding multiple copies of the same file across different folders is terribly inefficient. I would like be able to make soft links where I can have the 'appearence' of these files across different folders that would redirect to where the actual file is stored. Since the tablet runs on Linux this should be feasible. Has anyone tried this?

**Comment by Eeems** [comment]
The sync logic wouldn't really support this, and I don't think it's something that ddvk would really want to try to hack in and maintain between versions. That said, you probably could do some sort of deduping in the filesystem itself outside of the hacks with hardlinks.

https://remarkablewiki.com/tech/filesystem#user_data_directory_structure

**Comment by martin-braun** [comment]
Your best (and probably only) option is to use tags.

**Comment by eregen** [comment]
Yeah I've tried using the tags. I find them unwieldy and the remarkable brings them up whenever I open up any kind of doc. I've organized docs on my laptop with nested directories and many syslinks and that's what works best for me.

**Comment by dstango** [comment]
the file and directory structure is actually not a representation of the linux file and directory structure: each document gets a unique UUID and usually consists of several files: some meta-info in json, a .lines file for your strokes with the pen, and a .pdf file or .epub file if it's an ebook. Plus some directory for thumbnails or even cached rendered pages. (I'm not up to date about the most recent evolution and details of the file structure, but you get the idea.)
Thus if you'd want to have links, you'd need to create file system links for a bundle of files, plus create a modified copy of the metadata json-file that then contains the path you'd like it to appear in. It also can get "interesting" when files likes thumbnails or cached rendered versions of pages get created, as it might result in a multitude of copies.
To support such a thing with the hacks doesn't sound like something I would like to support if I were the developer, as it looks prone to brittleness with version changes.

Having said that I'd suggest you go forward to submit a feature request with the remarkable base product: adding a field with alternative paths in the metadata json file, and supporting these alternative paths in the document navigator doesn't sound too difficult to me, and would be the perfect place to have such a feature.

**Comment by martin-braun** [comment]
@dstango nailed it. reMarkable "file system" is just a database on another file system (Linux), so even if you manage to hack something together, there is no guarantee that it never break due to an update. While there might be a possibility to trick reMarkable to have a file in two folders (depending how the database structure actually works, idk), it's most likely super fragile, unsupported and can cause serious harm to your files or the entire device in case you update to a newer firmware.

I wouldn't advice to try this. You really want reMarkable to officially implement this, as @dstango said.

**Comment by eregen** [comment]
That's disappointing. I'll have to submit a feature request and just hope for the best. Thanks all for taking the time to  elucidate me.
